lucene/core/src/java/org/apache/lucene/util/automaton/Automata.java

```diff
@@ -85,7 +85,15 @@ public static Automaton makeAnyBinary() {
     a.finishState();
     return a;
   }
-  
+
+  /**
+   * Returns a new (deterministic) automaton that accepts all binary terms except
+   * the empty string.
+   */
+  public static Automaton makeAnyBinaryExceptEmpty() {
+    return makeCharRange(0, 255);
```

Comment:
I don't think this is right since this only accepts strings of length 1.

Reply:
Yes, I was missing that this doesn't include a self-referencing accept state. Fixed that and added more tests to check this is working now.